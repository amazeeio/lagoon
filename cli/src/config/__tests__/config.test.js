// @flow

import { createConfig, parseConfig } from '..';

const fs = require('../../util/fs');

// $FlowFixMe Jest can mutate exports https://github.com/facebook/jest/issues/936#issuecomment-214556122
fs.writeFile = jest.fn();

function _mock(fn: any): JestMockFn {
  return fn;
}

describe('createConfig', () => {
  it('should write default config to given path', async () => {
    createConfig('lagoon.yml', { project: 'your_project' });

    const [filename, data] = _mock(fs.writeFile).mock.calls[0];

    expect(filename).toBe('lagoon.yml');
    expect(data).toMatchSnapshot();
  });
});

describe('parseConfig', () => {
  it('should parse the config correctly', async () => {
    const yamlContent = await fs.readFile(`${__dirname}/lagoon.yml`);

    const config = parseConfig(yamlContent.toString());

    const expected = {
      project: 'amazee_io',
      deploy_tasks: {
        development: {
          before_deploy: ['cmd1', 'cmd2'],
          after_deploy: ['cmd1', 'cmd2'],
        },
        production: {
          before_deploy: ['cmd1', 'cmd2'],
          after_deploy: ['cmd1', 'cmd2'],
        },
      },
      shared: {
        production: [
          {
            src: 'files',
            dst: 'sites/default/files',
          },
        ],
      },
    };

    expect(config).toEqual(expected);
  });
});
