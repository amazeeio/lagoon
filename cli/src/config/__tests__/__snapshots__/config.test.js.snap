// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createConfig should write default config to given path 1`] = `
"project: your_project
deploy_tasks:
  task1:
    before_deploy: []
    after_deploy: []
"
`;
